third_party/ruby/t-none-context.patch

```diff
@@ -0,0 +1,146 @@
+diff --git gc.c gc.c
+index 6f4ae3a397..d4666b1406 100644
+--- gc.c
++++ gc.c
+@@ -6709,6 +6709,119 @@ gc_aging(rb_objspace_t *objspace, VALUE obj)
+ NOINLINE(static void gc_mark_ptr(rb_objspace_t *objspace, VALUE obj));
+ static void reachable_objects_from_callback(VALUE obj);
+ 
++/**
++ * BEGIN Stripe debugging code. 
++ *
++ * Context: We are seeing instances of "try to mark T_NONE" crashes at Stripe, at times tied to T_HASH parent object.
++ * In these cases, we want to print contextual information about the parent object at the time of the crash.
++ *
++*/
++typedef struct {
++  rb_objspace_t *objspace;
++  char *buff;
++  const int buff_size;
++  int* pos;
++} hash_debug_args_STRIPE;
++
++static int
++gc_debug_hash_entry_STRIPE(st_data_t key, st_data_t value, st_data_t args) {
++    #define BUFF_ARGS buff + *pos, buff_size - *pos
++    #define APPENDF(f) if ((*pos += snprintf f) >= buff_size) goto end
++    
++    // Destructure args
++    hash_debug_args_STRIPE *debug_args = (hash_debug_args_STRIPE*)args;
++    rb_objspace_t *objspace = debug_args->objspace;
++    char *buff = debug_args->buff;
++    int buff_size = debug_args->buff_size;
++    int *pos = debug_args->pos;
++
++    VALUE k_obj = (VALUE)key;
++    VALUE v_obj = (VALUE)value;
++
++    // Skip the entry that has T_NONEs, it probably caused the error condition.
++    // We care about surrounding context.
++    if (UNLIKELY(RB_TYPE_P(k_obj, T_NONE) || RB_TYPE_P(k_obj, T_NONE))) {
```

Comment:
Why `RB_TYPE_P(k_obj, T_NONE) || RB_TYPE_P(k_obj, T_NONE)` instead of `RB_TYPE_P(k_obj, T_NONE)`?

Reply:
Ah whoops, I meant v_obj, nice catch

Reply:
Fixed